src/test/java/com/google/jenkins/plugins/k8sengine/client/ContainerClientTest.java

```diff
@@ -16,108 +16,188 @@
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
+import static org.mockito.ArgumentMatchers.anyString;
 
 import com.google.api.services.container.Container;
+import com.google.api.services.container.Container.Projects.Zones.Clusters;
 import com.google.api.services.container.model.Cluster;
+import com.google.api.services.container.model.ListClustersResponse;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Optional;
-import org.junit.Before;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.mockito.Matchers;
-import org.mockito.Mock;
 import org.mockito.Mockito;
-import org.mockito.invocation.InvocationOnMock;
 import org.mockito.junit.MockitoJUnitRunner;
 import org.mockito.stubbing.Answer;
 
 /** Tests {@link ContainerClient}. */
 @RunWith(MockitoJUnitRunner.class)
 public class ContainerClientTest {
-  private ContainerClient containerClient;
-  @Mock private Container.Projects.Zones.Clusters clusters;
+  private static final String TEST_PROJECT_ID = "test-project-id";
+  private static final String OTHER_PROJECT_ID = "other-project-id";
+  private static final String ERROR_PROJECT_ID = "error-project-id";
+  private static final String EMPTY_PROJECT_ID = "empty-project-id";
+  private static final String TEST_ZONE = "us-west1-a";
+  private static final String OTHER_ZONE = "us-west2-b";
+  private static final String TEST_CLUSTER = "testCluster";
+  private static final String OTHER_CLUSTER = "otherCluster";
 
-  @Before
-  public void init() throws Exception {
+  private static ContainerClient containerClient;
+  private static Cluster testCluster;
+  private static Cluster otherCluster;
+
+  @BeforeClass
+  public static void init() throws Exception {
     Container container = Mockito.mock(Container.class);
-    ;
     Container.Projects projects = Mockito.mock(Container.Projects.class);
     Container.Projects.Zones zones = Mockito.mock(Container.Projects.Zones.class);
+    Clusters clusters = Mockito.mock(Container.Projects.Zones.Clusters.class);
     Mockito.when(container.projects()).thenReturn(projects);
     Mockito.when(projects.zones()).thenReturn(zones);
     Mockito.when(zones.clusters()).thenReturn(clusters);
-    this.containerClient = new ContainerClient(container);
-  }
+    testCluster = new Cluster().setName(TEST_CLUSTER).setZone(TEST_ZONE);
+    otherCluster = new Cluster().setName(OTHER_CLUSTER).setZone(TEST_ZONE);
 
-  @Test
-  public void testGetClusterReturnsProperlyWhenClusterExists() throws Exception {
-    Cluster testCluster = new Cluster();
-    testCluster.setName("testCluster");
-    testCluster.setZone("us-central1-c");
```

Comment:
Along similar vein as my feedback in the other test class, unit tests should avoid relying on central setUp methods to configure state they rely upon. An allowance for duplication of code is typically made in unit tests in order to provide more self-contained and readable tests.